tests/system/small/test_anywidget.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+
+import pytest
+
+import bigframes as bf
+
+
+def test_repr_anywidget_initial_state(
+    penguins_df_default_index: bf.dataframe.DataFrame,
+):
+    pytest.importorskip("anywidget")
+    with bf.option_context("display.repr_mode", "anywidget"):
+        from bigframes.display import TableWidget
+
+        widget = TableWidget(penguins_df_default_index)
+        assert widget.page == 0
+        assert widget.page_size == bf.options.display.max_rows
+        assert widget.row_count > 0
+
+
+def test_repr_anywidget_pagination_navigation(
+    penguins_df_default_index: bf.dataframe.DataFrame,
+):
+    """Test basic prev/next navigation functionality."""
+    pytest.importorskip("anywidget")
+    with bf.option_context("display.repr_mode", "anywidget"):
+        from bigframes.display.anywidget import TableWidget
+
+        widget = TableWidget(penguins_df_default_index)
+
+        # Test initial state
+        assert widget.page == 0
+
+        # Simulate next page click
+        widget.page = 1
+        assert widget.page == 1
```

Comment:
How do we tell if the data displayed has changed? Can we check the table HTML that it doesn't contain the first page of rows and that it does contain the second page?

Rather than the penguin data, it might be better to define our own DataFrame in this test and set a really small page size so we can make sure we visualize the correct rows. See: https://testing.googleblog.com/2018/02/testing-on-toilet-cleanly-create-test.html

> tests should never rely on default values that are specified by a helper method since that forces readers to read the helper method’s implementation details in order to understand the test.